### ISSUE TITLE
This will ever become a version 4 of knockout?

### ISSUE THREAD

**Issue by nmocruz** [knockout/tko#108]
Sorry by maybe put some pressure and be a bit selfish by not respecting the good work and effort made but seems to dying before becoming a real version 4 of knockout.
I fully understand the lack of time, I have also family and a job where the build of a framework is not so important as the client demands on features.

can something be done to help? how far is this to start a knockout version 4, or maybe tko beta?

**Comment by tscpp** [comment]
#1

**Comment by davidrwood** [comment]
It does seem to have died:

**Comment by nmocruz** [comment]
yes, last year zero activity I wonder if blazor will end like this, abandoned

**Comment by brianmhunt** [comment]
Not abandoned at all, still very actively used, but as it's being used in an early-stage company we're being extra careful about our support surface.  Hopefully positive things to report in the near-future. 🤞

**Comment by nmocruz** [comment]
near-future? sorry if it seems bold, but don't do like the aurelia team that promises something to next months and then stays 2 year releasing alpha versions. and that struggling to release a final version is very scary for us deciding what libraries to use.
I belive that if was 2 years ago I was migrating my projects to ko 4 without douts but now seems like all other frameworks are much powerful and feature-rich then ko. 
I recently started to use ko-jsx to solve performance issues, seems not bad at all. I hope that tko can release be released, 3 year ago was already to late to some people that moved way.

**Comment by Martlark** [comment]
I've found this current release to be excellent. Much easier and more productive than React.

**Comment by mattlacey** [comment]
@brianmhunt How large is the code base where it's being used? I know there's some work to do to move from 3.5x to TKO, but I am following with interest. We have a growing code code base, our ko components directory currently has around 5500 lines of code in html templates and 18,800 lines of JS in the view models for them (rough count in bash, nothing fancy). KO is still serving us well but keen to move to something more modern and I'm hoping TKO will be it :)

**Comment by brianmhunt** [comment]
@mattlacey Glad to hear your interest.  The current largest code-base for TKO (that I know of) is around 100 KO web-components over 100,000 lines of JSX+[JSS](https://cssinjs.org/?v=v10.6.0)+Typescript.

I'm hoping to round out some comfort for TKO as a drop-in replacement, plus a bunch of docs that describe why it's worthwhile and how to get started.  It's pretty close, but the big lingering item seems to be that TKO always uses its own CSP-safe parser `data-bind`, whereas knockout (without a plugin) uses `new Function('...')`.

Interestingly, Svelte seems to be converging on many of the things TKO does, which is validating.  They require a separate compiler though, whereas TKO uses standard & common JSX tools (which additionally makes bridging to/from React/Vue pretty trivial).  Great to see interest in the front-end development, with projects like Svelte.

For the immediate moment I'm tackling the strangely unsolved [build topology](https://github.com/knockout/tko/issues/134#issuecomment-849287766) problem.

**Comment by mattlacey** [comment]
JavaScript build systems are one of my least favourite things of all
time, though JavaScript packaging comes close.

When the time comes I'd be more than happy to do some tests on a branch
of our code base, to help gauge the level of effort involved and to help
try and find any snags/hangups. I've not swtiched away from KO because
it's not broken and works well for us, but it's always been on the back
of my mind as something that's going to need to be handled at some
stage.

On May 28, 2021, GitHub ***@***.***> wrote:
> @mattlacey <https://github.com/mattlacey> Glad to hear your interest.
> The current largest code-base for TKO (that I know of) is around 100
> KO web-components over 100,000 lines of JSX+JSS
> <https://cssinjs.org/?v=v10.6.0>+Typescript.
>
> I'm hoping to round out some comfort for TKO as a drop-in replacement,
> plus a bunch of docs that describe why it's worthwhile and how to get
> started. It's pretty close, but the big lingering item seems to be
> that TKO always uses its own CSP-safe parser data-bind, whereas
> knockout (without a plugin) uses new Function('...').
>
> Interestingly, Svelte seems to be converging on many of the things TKO
> does, which is validating. They require a separate compiler though,
> whereas TKO uses standard & common JSX tools (which additionally makes
> bridging to/from React/Vue pretty trivial). Great to see interest in
> the front-end development, with projects like Svelte.
>
> For the immediate moment I'm tackling the strangely unsolved build
> topology <https://github.com/knockout/tko/issues/134#issuecomment-
> 849287766> problem.
>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/knockout/tko/issues/108#issuecomment-850053181>, or
unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAJSWWYVP63BEFKRCBDMXITTP3ZCRANCNFSM4KR3RYBA>.

**Comment by nmocruz** [comment]
Congrats on version 4.0! One of the long-standing concerns about Knockout hasn’t only been modernization, but also performance when handling large numbers of objects. It might be worth highlighting performance improvements in TKO 4.0 as a key step forward for the framework.

I know that @ryansolid has some sympathy for Knockout and has worked on projects like ko-jsx, as well as experimenting with signals in other frameworks to improve performance. I think this is a promising direction, an upgrade path like this could help address performance issues in existing Knockout applications and reduce the need for full rewrites. 🚀

Strengthening this story could make TKO especially appealing for teams maintaining older Knockout projects.
